server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -726,10 +730,17 @@ public Pool createAndEnrichPools(Pool pool, List<Pool> existingPools) {
         List<Pool> pools = poolRules.createAndEnrichPools(pool, existingPools);
         log.debug("Creating {} pools: ", pools.size());
 
+        boolean isPoolLocked = pool.isLocked();
         for (Pool p : pools) {
             createPool(p);
         }
 
+        for (Pool tempPool : pools) {
+            if (isPoolLocked) {
```

Comment:
You can skip the declaration above and just query the value here. Since the locked flag is a simple boolean that is returned from the Pool object, the call gets optimized down to basically what you're doing here without needing to write it out yourself.

There are cases where you'll want to explicitly cache stuff locally, but since we're not dealing with unboxing and it's a simple accessor on what is little more than a data store, this is not such a case.

Reply:
Sure alright.